### PR TITLE
Adding configuration, mock data, environment variables

### PR DESCRIPTION
**Pull request by stephen1995**
image

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -1,27 +1,18 @@
-import { defineConfig } from 'cypress'
+import { defineConfig } from "cypress";
 
 export default defineConfig({
   e2e: {
     setupNodeEvents(on, config) {
       // if version not defined, use local
```
Comment (jalzaibar-advantagecg) [inline cypress.config.ts:6]: please avoid obvious code comments

File: cypress/e2e/interceptExample.cy.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+describe("Example for Intercept", () => {
+  const credentials = Cypress.env("credentials");
+  if (!credentials) {
+    throw new Error("Missing credentials for Local");
+  }
+
+  it("Mocking People from Contacts section - 1 person", () => {
+    cy.intercept("/contListResultsQuery.cfm", {
+      body: {
+        data: [
```
Comment (jalzaibar-advantagecg) [inline cypress/e2e/interceptExample.cy.ts:13]: please use fixtures to represent a set of data

File: cypress/e2e/interceptExample.cy.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { dashboardPage } from "../pages/dashboardPage";
+
+describe("Example for Intercept", () => {
+  const credentials = Cypress.env("credentials");
+  if (!credentials) {
+    throw new Error("Missing credentials for Local");
+  }
+
+  it("Mocking People from Contacts section - 1 person", () => {
+    cy.intercept("/contListResultsQuery.cfm", {
```
Comment (jalzaibar-advantagecg) [inline cypress/e2e/interceptExample.cy.ts:11]: please be explicit when using intercept. Declare http verb, path and use a fixture for the response.
https://docs.cypress.io/api/commands/fixture#Using-the-fixture-StaticResponse-property

File: cypress/pages/dashboardPage.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const dashboardPage = {
+  contactsDropdown: () => cy.xpath("//span[text()='Contacts']"),
```
Comment (jalzaibar-advantagecg) [inline cypress/pages/dashboardPage.ts:2]: please prefer to use IDs when referring page elements. If the page doesn't have one ask the developer to add it

File: cypress/pages/loginPage.ts [hunk 5]
```diff
@@ -1,17 +1,17 @@
-export const loginPage =  {
-   
+export const loginPage = {
+  usernameTxt: () => cy.get('[name="username"]'),
```
Comment (jalzaibar-advantagecg) [inline cypress/pages/loginPage.ts:2]: please avoid the use of unnecessary shortnames like txt, btn, chk, etc. Instead use full and descriptive names: userNameField. passwordField, logginButton, acceptCheckbox, etc

File: cypress/pages/loginPage.ts [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
 
-    usernameTxt: ()=> cy.get('[name="username"]'),
+  passwordTxt: () => cy.get('[name="pass"]'),
 
-    passwordTxt: ()=> cy.get('[name="pass"]'),
+  loginBtn: () => cy.get("#login"),
 
-    loginBtn: ()=>cy.get('#login'),
+  performLogin(username: string, password: string) {
+    //Verify there is no session available (Pending)
```
Comment (jalzaibar-advantagecg) [inline cypress/pages/loginPage.ts:9]: please ensure to create an item in Jira for all the pending work discovered, and start the comment with a TODO to standaryce handling

File: cypress/support/commands.js [hunk 7]
```diff
@@ -45,3 +45,7 @@ Cypress.Commands.add('downloadFile', (url, directory, fileName) => {
     })
   })
 
+  Cypress.Commands.add('customIntercept', (method, url, response) => {
+    cy.intercept(method, url, response);
```
Comment (jalzaibar-advantagecg) [inline cypress/support/commands.js:49]: Could you please elaborate why is this necessary?
wrapping a one line function with a command to receive the exact same parameters with no modifications looks likes is not adding value

File: cypress/support/e2e.ts [hunk 8]
```diff
@@ -14,16 +14,18 @@
 // ***********************************************************
 
 // Import commands.js using ES2015 syntax:
-import './commands'
+import "./commands";
+require("cypress-xpath");
 
 // Alternatively you can use CommonJS syntax:
```
Comment (jalzaibar-advantagecg) [inline cypress/support/e2e.ts:20]: please remove auto generated comments

File: .github/workflows/tests.yml [hunk 9]
```diff
@@ -13,12 +13,19 @@ jobs:
...
-        run: npm install
-
-      - name: Run tests
-        run: npm run local:chrome
+      - name: Cypress run
+        # Uses the official Cypress GitHub action https://github.com/cypress-io/github-action
+        uses: cypress-io/github-action@v6
+        with:
+          # Records to Cypress Cloud
+          # https://docs.cypress.io/guides/cloud/projects#Set-up-a-project-to-record
```
[the remark this answers is not in the document]
Comment (stephen1995) [reply .github/workflows/tests.yml:24]: Removing links for documentation

File: cypress.config.ts (hunk 1 above)
Comment (stephen1995) [reply cypress.config.ts:6]: Removing comment

File: cypress/e2e/interceptExample.cy.ts (hunk 2 above)
Comment (stephen1995) [reply cypress/e2e/interceptExample.cy.ts:13]: Updated using onePerson json file.

File: cypress/e2e/interceptExample.cy.ts (hunk 3 above)
Comment (stephen1995) [reply cypress/e2e/interceptExample.cy.ts:11]: Done in previous comment

File: cypress/pages/dashboardPage.ts (hunk 4 above)
Comment (stephen1995) [reply cypress/pages/dashboardPage.ts:2]: Its obvious that it didn't have ID that's why I used xpath.

File: cypress/pages/loginPage.ts (hunk 5 above)
Comment (stephen1995) [reply cypress/pages/loginPage.ts:2]: Updating and adding this to confluence convention

File: cypress/pages/loginPage.ts (hunk 6 above)
Comment (stephen1995) [reply cypress/pages/loginPage.ts:9]: Comment updated

File: cypress/support/commands.js (hunk 7 above)
Comment (stephen1995) [reply cypress/support/commands.js:49]: Command removed.

File: cypress/support/e2e.ts (hunk 8 above)
Comment (stephen1995) [reply cypress/support/e2e.ts:20]: Those comments come from Cypress xD.

**Review by jalzaibar-advantagecg — APPROVED** [review]
looks good now

**Review by jalzaibar-advantagecg — APPROVED** [review]

**Review by jalzaibar-advantagecg — APPROVED** [review]
